google-cloud-firestore/lib/google/cloud/firestore/query_partition.rb

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # QueryPartition
+      #
+      # Represents a split point that can be used in a query as a starting and/or end point for the query results.
+      #
+      # The cursors returned by {#start_at} and {#end_before} can only be used in a query that matches the constraint of
+      # the query that produced this partition.
+      #
+      # See {CollectionGroup#partitions} and {Query}.
+      #
+      # @!attribute [r] start_at
+      #   The cursor values that define the first result for this partition, or `nil` if this is the first partition.
+      #   Returns an array of values that represent a position, in the order they appear in the order by clause of the
+      #   query. Can contain fewer values than specified in the order by clause. Will be used in the query returned by
+      #   {#create_query}.
+      #   @return [Array<Object>, nil] Typically, the values are {DocumentSnapshot} objects.
```

Comment:
I am not sure I fully follow what the return value here refers to, but if it is the cursor value then it should be an array of field values and not a DocumentSnapshot.

Reply:
Thanks for catching this. I changed `DocumentSnapshot` to `DocumentReference` in this description.